api/src/paths/dwc/eml.test.ts

```diff
@@ -0,0 +1,662 @@
+import chai, { expect } from 'chai';
+import { describe } from 'mocha';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import { getMockDBConnection } from '../../__mocks__/db';
+import * as db from '../../database/db';
+import { CustomError } from '../../errors/CustomError';
+import * as eml from './eml';
+import * as eml_queries from '../../queries/dwc/dwc-queries';
+import SQL from 'sql-template-strings';
+
+chai.use(sinonChai);
+
+const dbConnectionObj = getMockDBConnection({
+  systemUserId: () => {
+    return 20;
+  }
+});
+
+describe('getSurveyDataPackageEML', () => {
+  const sampleReq = {
+    keycloak_token: {},
+    body: {
+      data_package_id: 1
+    }
+  } as any;
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should throw a 400 error when no data package id is provided', async () => {
+    sinon.stub(db, 'getDBConnection').returns(dbConnectionObj);
+
+    try {
+      const result = eml.getSurveyDataPackageEML();
+      await result(
+        { ...sampleReq, body: { ...sampleReq.body, data_package_id: null } },
+        (null as unknown) as any,
+        (null as unknown) as any
+      );
+      expect.fail();
+    } catch (actualError) {
+      expect((actualError as CustomError).status).to.equal(400);
+      expect((actualError as CustomError).message).to.equal('Missing required body param `data_package_id`.');
```

Comment:
Same as the above comment, most of our existing tests check status before message.  But I want to start changing it so that the message assertion comes before the status assertion.  Because otherwise, if the wrong error is thrown (like a 500), all you get in the log is "400 !== 500" which is super unhelpful when trying to debug the test, since there might be multiple places where a 500 could be thrown from.